pytradfri/device.py

```diff
@@ -216,10 +219,9 @@ def set_kelvin_color(self, kelvins, *, index=0):
     def set_predefined_color(self, colorname, *, index=0):
         try:
             color = COLORS[colorname.lower().replace(" ", "_")]
-        except:
-            pass
-        else:
             return self.set_hex_color(color, index=index)
+        except KeyError:
```

Comment:
I don't like swallowing errors. Because now as a caller, I won't know if my call succeeded and that means that I can't communicate this to my users. I think that we should never catch but instead raise as much as possible. That way we leave it to developers that use this lib if they want to swallow errors or show them to users.